runtime/generator.go

```diff
@@ -44,6 +44,7 @@ type Generator struct {
 
 // NewGenerator returns a new Generator object that runs the given Block b.
 func NewGenerator(f *Frame, fn func(*Object) (*Object, *BaseException)) *Generator {
+	f.taken = true // Claim the frame from being returned.
```

Comment:
For now, I think it makes sense to have `taken` be viral up the frame stack. This helps in the cases where the traceback is grabbed, and the frame data is inspected later. It isn't optimal, but it strikes a balance between speed and memory pressure. Eventually the GC will help out.

Now, that `__frame__()` thing is scary...and I didn't know about it. I'll taint the frame in this change, but I think the method could be removed, and replaced with something a little less leaky.